Automattic/jetpack
# Jetpack AI: open upgrade path on new tab, part 2

Part of #37458.

This change will make the upgrade button from the upgrade prompt/nudge opens on a new tab instead of redirecting the editor tab.

## Proposed changes:

* Add support for `target` on the shared `Nudge` component and the internal `LightNudge` component
* Change target of buttons and links to `_blank`
* Remove calls to `autoSaveAndRedirect` hook
* Add missing `placement` property on the upgrade prompt from the AI Featured Image Generator tool

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Run this branch on your testing site
* Make sure you reach the requests limit of the testing site:
   * you can use up to 20 requests on a fresh, new JN site, OR
   * you can sandbox the `public-api` and then update the `0-sandbox.php` file with the following filters (this will enable the 100-requests plan and set it to maximum usage, so your site requires an upgrade):

```

add_filter( 'jetpack_ai_current_period_requests_count', function() { return 100; } );
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 100; } );
```

* Go to the block editor, open the browser's console and enable debugging with `localStorage.setItem( 'debug', '*' );`
* Insert an AI Assistant block; you will see the upgrade prompt, as usual:

<img width="600" alt="Screenshot 2024-05-27 at 12 57 17" src="https://github.com/Automattic/jetpack/assets/6760046/c6f1159a-2098-45c8-a45e-d4cb8d26a3ca">

* Click the upgrade button
   * Confirm it will open on a new tab
   * Confirm the tracking event is fired on the original tab, by checking the logs on the console:

<img width="600" alt="Screenshot 2024-05-27 at 12 59 01" src="https://github.com/Automattic/jetpack/assets/6760046/11587d82-724b-4e22-a425-e7ca4762a5c3">

* Now look for the featured image section on the settings sidebar; it can be on a dedicated section, or inside the summary section
* Click "Set featured image" and then select "Generate with AI" to trigger the AI Featured Image generator; you will see the light upgrade prompt, as usual:

<img width="600" alt="Screenshot 2024-05-27 at 12 57 38" src="https://github.com/Automattic/jetpack/assets/6760046/3cf23e0a-ce95-4ad0-b759-dc514bd7f171">

* Click the upgrade link
   * Confirm it will open on a new tab
   * Confirm the tracking event is fired on the original tab, by checking the logs on the console
* Now look for the excerpt section on the settings sidebar; you will see the upgrade prompt, as usual:

<img width="250" alt="Screenshot 2024-05-27 at 12 56 45" src="https://github.com/Automattic/jetpack/assets/6760046/462cb82c-b06c-4efe-8445-9890bd96e51c">

* Click the upgrade button
   * Confirm it will open on a new tab
   * Confirm the tracking event is fired on the original tab, by checking the logs on the console
